go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -412,15 +404,52 @@ func (c *Controller) syncServices(key string) error {
 
 	// at this point we have processed all vips we've found in the service
 	// so the remaining ones that we had in the vipsTracked variable should be deleted
-	err = deleteVIPsFromOVN(vipsTracked, c.serviceTracker, name, namespace, c.clusterPortGroupUUID)
+	// We remove them from OVN and from the tracker
+	err = deleteVIPsFromOVN(vipsTracked, c.serviceTracker, name, namespace)
 	if err != nil {
 		c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToDeleteOVNLoadBalancer",
 			"Error trying to delete the OVN LoadBalancer for Service %s/%s: %v", name, namespace, err)
 		return err
 	}
+	c.serviceTracker.deleteServiceVIPs(name, namespace, vipsTracked)
 	return nil
 }
 
+func (c *Controller) addServiceToIdlingBalancer(vips sets.String, service *v1.Service) error {
+	for _, vipProtocol := range vips.List() {
+		vip, protocol := splitVirtualIPKey(vipProtocol)
+		lb, err := loadbalancer.GetOVNKubeIdlingLoadBalancer(protocol)
+		if err != nil {
+			return errors.Wrapf(err, "Error getting OVN idling LoadBalancer for protocol %s", protocol)
+		}
+		err = loadbalancer.UpdateLoadBalancer(lb, vip, []string{})
+		if err != nil {
+			return errors.Wrapf(err, "Failed to update idling loadbalancer")
+		}
+		// update the tracker with the VIP
+		c.serviceTracker.updateService(service.Name, service.Namespace, vip, protocol)
+	}
+	return nil
+}
+
+func getNodeportIPs() ([]string, error) {
```

Comment:
instead of this do you want to use utils.go func getNodeIPs(isIPv6 bool) ([]string, error) {

It basically does the same thing as this function, but does it per IP family. We need to make sure ip families match for dualstack.

Reply:
But this is used where I collect all the addresses to move, so I guess we don't care if they are v4 or v6.
I can use the other function twice anyway.